File: src/test/java/eu/bitwalker/useragentutils/browser/InternetExplorerXboxParameterizedTest.java
package eu.bitwalker.useragentutils.browser;

import static java.util.Arrays.asList;

import java.util.Collection;

import org.junit.runners.Parameterized.Parameters;

import eu.bitwalker.useragentutils.Browser;
import eu.bitwalker.useragentutils.OperatingSystem;

public class InternetExplorerXboxParameterizedTest extends AbstractUserAgentParameterizedTest {

	public InternetExplorerXboxParameterizedTest(String userAgentValue,
			Browser expectedBrowser, String browserVersionExpected,
			OperatingSystem osExpected) {
		super(userAgentValue, expectedBrowser, browserVersionExpected,
				osExpected);
	}

	@Parameters
	public static Collection<Object[]> testData() {
		return asList(new Object[][] {

				{
						"Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; Trident/6.0; Xbox; Xbox One)",
						Browser.IE_XBOX, "10.0", OperatingSystem.XBOX_OS },
				{
						"Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0; Xbox)",
						Browser.IE_XBOX, "9.0", OperatingSystem.XBOX_OS }
		});
	}
}

File: src/test/csharp/eu/bitwalker/useragentutils/browser/InternetExplorerXboxParameterizedTest.cs
using System;
using NUnit.Framework;

namespace  eu.bitwalker.useragentutils.browser {


public class InternetExplorerXboxParameterizedTest : AbstractUserAgentParameterizedTest {

				[TestCase(
						"Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; Trident/6.0; Xbox; Xbox One)",
						"IE_XBOX", "10.0", "XBOX_OS" )]
				[TestCase(
						"Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0; Xbox)",
						"IE_XBOX", "9.0", "XBOX_OS" )]
    [Test]
    public void testData(String userAgentValue,
            string expectedBrowserStr, String expectedBrowserVersion,
            string expectedOSStr)
    {
        shouldParseUserAgent(userAgentValue, Browser.valueOf(expectedBrowserStr), expectedBrowserVersion, OperatingSystem.valueOf(expectedOSStr));
    }
}
}
